dpcpp/test/matrix/sellp_kernels.cpp

```diff
@@ -0,0 +1,398 @@
+        alpha = gko::initialize<Vec>({2.0}, ref);
+        beta = gko::initialize<Vec>({-1.0}, ref);
+        dmtx = Mtx::create(dpcpp);
+        dmtx->copy_from(mtx.get());
+        dempty = Mtx::create(dpcpp);
+        dresult = Vec::create(dpcpp);
+        dresult->copy_from(expected.get());
+        dy = Vec::create(dpcpp);
+        dy->copy_from(y.get());
+        dalpha = Vec::create(dpcpp);
+        dalpha->copy_from(alpha.get());
+        dbeta = Vec::create(dpcpp);
+        dbeta->copy_from(beta.get());
+    }
+
+    std::shared_ptr<gko::ReferenceExecutor> ref;
+    std::shared_ptr<const gko::DpcppExecutor> dpcpp;
+
+    std::ranlux48 rand_engine;
+
+    std::unique_ptr<Mtx> mtx;
+    std::unique_ptr<Mtx> empty;
+    std::unique_ptr<Vec> expected;
+    std::unique_ptr<Vec> y;
+    std::unique_ptr<Vec> alpha;
+    std::unique_ptr<Vec> beta;
+
+    std::unique_ptr<Mtx> dmtx;
+    std::unique_ptr<Mtx> dempty;
+    std::unique_ptr<Vec> dresult;
+    std::unique_ptr<Vec> dy;
+    std::unique_ptr<Vec> dalpha;
+    std::unique_ptr<Vec> dbeta;
+};
+
+
+TEST_F(Sellp, SimpleApplyIsEquivalentToRef)
+{
+    set_up_apply_matrix();
+
+    mtx->apply(y.get(), expected.get());
+    dmtx->apply(dy.get(), dresult.get());
+
+    auto result = Vec::create(ref);
+    result->copy_from(dresult.get());
+    GKO_ASSERT_MTX_NEAR(result, expected, r<vtype>::value);
+}
+
+
+TEST_F(Sellp, AdvancedApplyIsEquivalentToRef)
+{
+    set_up_apply_matrix();
+
+    mtx->apply(alpha.get(), y.get(), beta.get(), expected.get());
+    dmtx->apply(dalpha.get(), dy.get(), dbeta.get(), dresult.get());
+
+    auto result = Vec::create(ref);
+    result->copy_from(dresult.get());
+    GKO_ASSERT_MTX_NEAR(result, expected, r<vtype>::value);
+}
+
+
+TEST_F(Sellp, SimpleApplyWithSliceSizeAndStrideFactorIsEquivalentToRef)
+{
+    set_up_apply_matrix(1, 32, 2);
+
+    mtx->apply(y.get(), expected.get());
+    dmtx->apply(dy.get(), dresult.get());
+
+    auto result = Vec::create(ref);
+    result->copy_from(dresult.get());
+    GKO_ASSERT_MTX_NEAR(result, expected, r<vtype>::value);
+}
+
+
+TEST_F(Sellp, AdvancedApplyWithSliceSizeAndStrideFActorIsEquivalentToRef)
+{
+    set_up_apply_matrix(1, 32, 2);
+
+    mtx->apply(alpha.get(), y.get(), beta.get(), expected.get());
+    dmtx->apply(dalpha.get(), dy.get(), dbeta.get(), dresult.get());
+
+    auto result = Vec::create(ref);
+    result->copy_from(dresult.get());
+    GKO_ASSERT_MTX_NEAR(result, expected, r<vtype>::value);
+}
+
+
+TEST_F(Sellp, SimpleApplyMultipleRHSIsEquivalentToRef)
+{
+    set_up_apply_matrix(64);
+
+
+    mtx->apply(y.get(), expected.get());
+    dmtx->apply(dy.get(), dresult.get());
+
+    auto result = Vec::create(ref);
+    result->copy_from(dresult.get());
+    GKO_ASSERT_MTX_NEAR(result, expected, r<vtype>::value);
+}
+
+
+TEST_F(Sellp, AdvancedApplyMultipleRHSIsEquivalentToRef)
+{
+    set_up_apply_matrix(64);
+
+
+    mtx->apply(alpha.get(), y.get(), beta.get(), expected.get());
```

Comment:
Same here (and a lot other lines in this file):
```suggestion
    set_up_apply_matrix(64);

    mtx->apply(alpha.get(), y.get(), beta.get(), expected.get());
```